### ISSUE TITLE
Interest in a creating a physical geography companion deck?

### ISSUE THREAD

**Issue by ScottON** [axelboc/anki-ultimate-geography#358]
I'm nearing completion of the Ultimate Geography deck. Firstly, thank you for investing the time to make this resource available - I can't praise it enough. Secondly, I've started thinking about what I want to utilise that additional time for going forward.

I'm leaning towards iconic and noteworthy physical geography - mountain ranges, deserts, rivers, lakes, mountains, and volcanoes. I've loosely thought about inclusion criteria but I'll hold off on discussing that until I've gauged interest.

Before I begin creating cards I was curious if any of the contributors to this deck would be interested in helping/leading the project? I've made plenty of cards but never managed a collaborative project like this before and would prefer to pass the organisational side to someone else. Also wanted to suggest that if there is interest in the project that it's shared on /r/anki with a request for contributors while it's in its infancy. Ultimate Geography is held in very high regard there and I'm sure the project would be well received as long as there is some guidance regarding GitHub.

I'm interested in hearing your thoughts.  Thanks!

**Comment by ScottON** [comment]
Thanks for taking the time to respond. Great to see that there is already some interest and discussion on this.

> If you don't want to manage a separate deck, perhaps we could get the ball rolling on this here at UG?

If it's something you're going to be doing eventually then that works with me. I'd anticipated this post getting more responses. If there are other contributors who would work on a shared deck I'd be delighted to contribute. However if it's going to be largely driven by myself then I will probably just make cards for my personal deck as I don't need to worry about copyright issues on any resources I use and can make cards that reference my current knowledge.

>How awesome would it be if we could make decks composable

This would be fantastic! Assuming there was a big enough library of notes I could see something like that driving adoption of Anki in schools and universities. The only real issue I can see is people getting FOMO and wanting to learn **all the things!!!**. I know that I would get a tad carried away.

> I had no idea UG was being talked about on Reddit!! 

It's definitely one of the most popular decks. If you are ever looking for more contributors/have some spare time I'd highly recommend doing an AMA in /r/Anki. You'd get more people using the deck, more contributors, and possibly inspire some other community projects.

Once again, I appreciate you taking the time to respond. I'll wait a few weeks and see if this post receives gains any traction before deciding how to proceed. Take care!

**Comment by aplaice** [comment]
As @axelboc wrote, thanks very much for the kind words and for your enthusiasm!

Slightly rough (and belated) thoughts below:

I'd be very happy to participate in the creation of a physical geography (sub?)deck, though I don't think I'll have time to focus on this. (I've been planning to help along with CrowdAnki and BrainBrew, though still haven't got around to it...)

<hr/>

> The only real issue I can see is people getting FOMO and wanting to learn all the things!!!.

Yeah, I totally get this feeling! :) 

<hr/>

Regarding asking for contributors, I think that it'd be best if we waited until AUG switches to Brain Brew (I guess that irrespective of whether the physical geography deck is developed in this repo or not, it'd make sense for it to rely on the same general infrastructure/tooling). That's definitely not to say that development should stop, just that actively advertising for new people might be a waste now, since they'd have to get acquainted with the current system (and install PHP etc....) unnecessarily.

If you have ideas (even rough ones) about inclusion criteria, then please post them here, (ideally as separate issues)! If necessary, the discussion can be linked or copy-pasted over to a new repo. (I don't think there's a good way to migrate issues between repositories on GitHub :/.)

On the question of having a separate repository or not, I can see various arguments in favour of either approach:

* We already have seas and islands, which are physical geography, here.

* People might be interested in only physical or only political geography.

* If the new (not-seas-and-islands) physical geography section grows quickly/is initially unstable, it might be best to keep it separate, at least at the start.

* Since splitting and merging git history (even branches without a shared ancestor) is trivial, at the cost of a slightly bloated `.git`, we can change our minds at any point.

<hr/>

In the meantime, obviously please don't feel under pressure not to work on your own notes for the relevant topics, etc., if you ever feel that we're going too slow here!

**Comment by Flippo008** [comment]
@ScottON 

I am in a quite similar position as you, wondering how could I expand my geography knowledge. 

I found another well-made geography deck you could find helpful called **Western Geography**. It is not very known (only one upvote and that from me) and in a comment, made two months ago, I suggested he could add more content. Currently, it has Candian, German etc. territories and states, in addition to **lakes and rivers**. However, the last update was in 2017 so I don't think this deck is updated. (I might be wrong here) I would also like to note that individual provinces, territories and states could be added in Ultimate Geography as well. 
https://ankiweb.net/shared/info/1873986475

Mainly I wanted to say that I would love to see complex anki deck just for physical geography and would really appreciate if one was made.

**Comment by ScottON** [comment]
@aplaice  waiting till Brain Brew is ready before starting a 'requirement drive'  makes sense. 

>If you have ideas (even rough ones) about inclusion criteria, then please post them here, (ideally as separate issues)!

I'll get that done in the next wee while. I've just been told I'm going back to work in September so I'm not going to have as much free time available as I'd expected.

>If the new (not-seas-and-islands) physical geography section grows quickly/is initially unstable, it might be best to keep it separate, at least at the start.

I've got very limited experience with GitHub so I don't know if it could be developed in this repository and assembled as a separate deck? There are already a pool of experienced contributors here and it would be a shame to lose any of the talent by splitting the project. If it is not possible to do that then I'm in favour of having a separate repository as UG is already a sizeable deck and adding a lot of new cards will only make it more daunting.

@Flippo008 thanks for sharing that link and expressing your interest. I've only had a quick glance but it looks like that covers some of the information I'm interested in learning.
> I would also like to note that individual provinces, territories and states could be added in Ultimate Geography as well.

As I understand it this might become possible in the future if they manage to develop composable decks at Brain Brew. The issue with that right now is where do you draw the line? If you include everything the deck will become bloated and impractical for all but the most devout students. If you choose to only include X,Y,Z then the remainder of the 'alphabet' is offended. 

You might also enjoy this deck for [United Kingdom (UK) Geography: Regions, Counties, and Cities](https://ankiweb.net/shared/info/80961363)

**Comment by ohare93** [comment]
Fyi, I have just completed my major 3+ month task of restructuring Brain Brew and have [released a new version](https://github.com/ohare93/brain-brew/releases/tag/v0.2.1).

With that done, I am now capable of switching over UG to Brain Brew :+1: This update fixes a bunch of issues from before, though it would still be using Csvs for now (though I was tempted to instead immediately start on Yaml source files :wink: @axelboc), and Note Models are not auto generated (yet), but it's all coming together! :grin: 

If I make a working repo next weekend, will folks give it a perusal? :pray: If so, I will take that feedback and decide what to focus on going forward from there :+1:

**Comment by axelboc** [comment]
Awesome!

> If I make a working repo next weekend, will folks give it a perusal? 

For sure! Looking forward to it 🙌

**Comment by axelboc** [comment]
Alright, closing this one as well, since there's nothing actionable in UG for the time being. Feel free to keep commenting, of course. 😉

**Comment by Flippo008** [comment]
@aplaice 
I wanted to ask you about this for some time, hoping this is a relevant topic. I am going to make my personal deck of the Czech Republic containing its districts, mountains, rivers etc. I would like to have similar images of maps like these used in UG, however, I do not know how to make those. I noticed you are the one making these maps in UG and perhaps you could help me. Where do you find "source" of the map where you highlight a certain country or sea? How do you make the highlight or zoom box? Is there a specific program or website, or do you use photoshop/illustrator? With this information, I could contribute to making physical geography companion deck in future.

Thanks in advance.

**Comment by ohare93** [comment]
I have this noted down, perhaps I can quickly help and @aplaice can correct if needed 👏 from comments here https://github.com/axelboc/anki-ultimate-geography/issues/1#issuecomment-552244462 and here https://github.com/axelboc/anki-ultimate-geography/pull/176 about it, and it is called Mapshaper which you can find at https://github.com/mbloch/mapshaper/ I have never used it, but it seems rather straightforward from what I can see. I hope it can help you.

I would also be interested in a more concrete breakdown of how you go about generating the maps, btw 👏 😁 @aplaice

**Comment by aplaice** [comment]
> I wanted to ask you about this for some time, hoping this is a relevant topic. I am going to make my personal deck of the Czech Republic containing its districts, mountains, rivers etc. I would like to have similar images of maps like these used in UG, however, I do not know how to make those. I noticed you are the one making these maps in UG and perhaps you could help me. Where do you find "source" of the map where you highlight a certain country or sea? How do you make the highlight or zoom box? Is there a specific program or website, or do you use photoshop/illustrator? With this information, I could contribute to making physical geography companion deck in future.

To a large extent, it depends on the map.

(This turned out much longer than expected; I hope it's useful.)

## Modify upstream

For some of them, I just adjust the existing maps on Wikimedia. 

For example, for [the map of Scandinavia](https://commons.wikimedia.org/wiki/File:Scandinavia_in_its_region.svg), I took an existing map [of Norway](https://commons.wikimedia.org/wiki/File:Norway_in_its_region.svg) and changed the highlighting (this was a matter of changing the styling of the elements of [the SVG](https://upload.wikimedia.org/wikipedia/commons/1/14/Scandinavia_in_its_region.svg) (in this case the "`fill`") — I sometimes use Inkscape for this, sometimes I open the SVG in Firefox and use `Inspect Element` and the debugger toolbar tooling (and then save the modified file) and sometimes, when I'm doing a very similar modification yet another time, in a text editor).

In other cases, the changes are more extensive, for instance when combining multiple maps (from Wikimedia), for zoomed-in-areas. I do that in Inkscape or copy-paste the SVG from one file into the other, modifying the source by hand, (or some combination of the two). The zoom boxes themselves are a couple of rectangles and polygons. (I also have a small set of scripts for helping editing, cropping correctly etc., but they're extraordinarily messy, they're adapted to my idiosyncrasies and I change them on-the-go, so sharing them would not be useful.) I'd probably recommend simply using Inkscape.

(I don't use Inkscape quite as often as I should, since Inkscape usually significantly changes the internal formatting of the SVG files, while I like minimal, clean diffs to make sure that I didn't change anything that I shouldn't have. (The changes to the internal formatting don't have any negative effects, so it's just a weird aesthetic preference on my part.) If you don't tend to over-worry, like I do, then Inkscape should be more-or-less ideal.)

(Photoshop or the like would also probably work, but I've never used it.)


tldr;

1. Look at some of the many existing maps on Wikimedia, and inspect the SVG files themselves with your browser's debugging tools (inspect element) for inspiration regarding styling etc. and as a base for modification.

2. Inkscape (or any other vector editor).


### Generating from scratch

When generating from scratch, you need the source data (one or more "shapefiles", usually in `.shp` or `.geojson`/`.json` format), and a program to "project" the data and convert into an SVG.

#### Data

I usually get the data from [Natural Earth](https://www.naturalearthdata.com/downloads/).

For districts of the Czech republic, you'd probably need some of the data [here](https://www.naturalearthdata.com/downloads/10m-cultural-vectors/10m-admin-0-details/), though it might not be fine/precise enough.

For rivers, you'd need something like [this](https://www.naturalearthdata.com/downloads/10m-physical-vectors/10m-rivers-lake-centerlines/), though again I'm not sure if they have sufficiently detailed data (having all of the smaller rivers etc.).

#### Conversion

As @ohare93 noted, I use [`mapshaper`](https://github.com/mbloch/mapshaper/) for the conversion. 

(I'm assuming a Unix-like environment (Linux/MacOS/WSL) for the following, but I'm pretty sure that it should work on Windows (in Powershell or in a `cmd` prompt), as well, with little modification.)

You can install it "globally" (the binary becomes part of your "path") with:

```bash
npm install -g mapshaper
```

or locally (in the directory/folder corresponding to your project) with

```bash
npm install mapshaper
```

If you install locally, the "binaries"/programs are available at `node_modules/.bin/` within the folder in which you ran `npm install ...`.

`mapshaper-gui` is really convenient particularly for visualising and inspecting the data. 

`mapshaper` can be used non-interactively to run the conversion, for easy reproducibility. (For example:

```bash
mapshaper ne_your_shapefile.shp -proj '+proj=wintri +lon_0=0' -o the_output.svg
# or (depending on if you installed locally or globally)
node_modules/.bin/mapshaper ne_your_shapefile.shp -proj '+proj=wintri +lon_0=0' -o the_output.svg
```

will project the data in `ne_your_shapefile.shp` according to the [Winkel Tripel projection](https://en.wikipedia.org/wiki/Winkel_tripel_projection) (I usually choose that since it's what is typically used by Wikimedia's ["Map Workshop"](https://en.wikipedia.org/wiki/Wikipedia:Graphics_Lab/Map_workshop)) and convert it into an image, in the file `the_output.svg`.)

`mapshaper` has useful documentation, starting [here](https://github.com/mbloch/mapshaper/wiki/Introduction-to-the-Command-Line-Tool).

Some alternatives to `mapshaper` are QGis or ArcGIS.

If you want to have mini-maps, or the like, you might need to combine several SVG maps. In this case, you're probably back to using an image editor like Inkscape.

(I have a script for choosing the right styling, combining the main map with the mini map and highlighting the desired region, which I used for the sea maps, [here](https://github.com/aplaice/svg_sea_maps/), but it's horribly over-complicated and probably over-engineered, as it grew while I was getting acquainted with `mapshaper` so I don't recommend using or even looking at it. (Possibly, the various options available in `mapshaper`, [that I used](https://github.com/aplaice/svg_sea_maps/blob/master/main_map.sh#L165-L171) might be marginally useful, for reference/comparison/as an example, but nothing else.))

tldr;

1. Source of shapefiles. e.g. [Natural Earth](https://www.naturalearthdata.com/downloads/), if it's fine-grained enough.

2. Conversion software. e.g. [`mapshaper`](https://github.com/mbloch/mapshaper/).

<hr/>

I hope that this is at least slightly useful!

**Comment by Flippo008** [comment]
Thank you very much for a long and comprehensive answer. I will go through this advice and decide what is best for me. 

In case of using pictures available online (google images), is there any copyright issue to worry about? @ScottON mentioned it above.

> However if it's going to be largely driven by myself then I will probably just make cards for my personal deck as I don't need to worry about copyright issues on any resources I use and can make cards that reference my current knowledge.

E.g. I download a google image and edited with photoshop (I have some experience with it, so it is more convenient for me) colours and highlighted certain areas. Is it part of **fair use**?

**Comment by aplaice** [comment]
> E.g. I download a google image and edited with photoshop (I have some experience with it, so it is more convenient for me) colours and highlighted certain areas. Is it part of fair use?

(Disclaimer: I am not a lawyer. etc.) If you're planning to use it for your personal use or share it privately (i.e. not as a public deck on Ankiweb) with a small number of friends, then it's fair use, both morally and (to the extent I, as a non-lawyer can tell) legally.

Perhaps more importantly, if you're not planning to publicly share the images, then how would the copyright holders know and why would they care? :D
